core/src/apps/cardano/layout.py

```diff
@@ -394,6 +397,52 @@ async def confirm_withdrawal(
     await require_confirm(ctx, page1)
 
 
+async def confirm_catalyst_registration(
+    ctx: wire.Context,
+    public_key: str,
+    staking_path: List[int],
+    reward_address: str,
+    nonce: int,
+) -> None:
+    pages: List[ui.Component] = []
+
+    page1 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+    page1.bold("Confirm voting key")
+    page1.bold("registration.")
+    page1.normal("Nonce: %s" % nonce)
```

Comment:
I wouldn't put the nonce on the first screen, it's arguably the least relevant attribute and it can be confusing for users to see that right at the beginning before more relevant things they would otherwise expect, like the voting key or staking key info. The nonce imho belongs somewhere around the end of the flow, even at the cost of adding one more screen.

Reply:
Ok, I agree. I just wanted to save the one extra screen at the end. Done in https://github.com/vacuumlabs/trezor-firmware/pull/57/commits/02a3b6f62c4f15bba6d5ed2f6ef0be7c19ee17e6.